src/langapi/mode.cpp

```diff
@@ -30,6 +30,17 @@ int get_mode(const std::string &str)
   return -1;
 }
 
+int get_old_frontend_mode(int current_mode)
+{
+  unsigned i;
+  auto expected = mode_table[current_mode++].name;
+  for(i = current_mode; mode_table[i].name != nullptr; i++)
+    if(expected == mode_table[i].name)
```

Comment:
This comparison on `const char *` relies on string literals getting the same address in the translation unit `src/esbmc/globals.cpp`. This is [not guaranteed](https://en.cppreference.com/w/cpp/language/string_literal):
> Whether string literals can overlap and whether successive evaluations of a string-literal yield the same object is unspecified. That means that identical string literals may or may not compare equal when compared by pointer.

Reply:
Nice catch. I will use `std::string` then.